include/cuco/detail/open_addressing/open_addressing_impl.cuh

```diff
@@ -705,7 +705,8 @@ class open_addressing_impl {
       }
     };
 
-    cub::DeviceFor::ForEachCopyN(storage_ref.data(), storage_ref.num_windows(), op, stream.get());
+    CUCO_CUDA_TRY(cub::DeviceFor::ForEachCopyN(
```

Comment:
`CUCO_CUDA_TRY` will fetch the last CUDA error and thus synchronize the device with the host. We can't do that in a function marked `_async`. The error will pop up anyway if we call the non-'_async` variant after we call `stream.wait()`. For the `_async` variant, the user is in charge of proper CUDA error handling.

Reply:
> fetch the last CUDA error and thus synchronize the device with the host. 

IIUC, it only synchronizes when the return value is not `cudaSuccess`. And in the case it throws, whether it synchronizes or not no longer matters.
https://github.com/NVIDIA/cuCollections/blob/356ce59bb9a403437bf7b1be668bc914ad2bf26f/include/cuco/detail/error.hpp#L54-L58

Reply:
Hmm, good point. Just to make sure I got this right: The cub/kernel call will immediately return and will not wait for the kernel to finish? Then we should probably wrap all kernel launches in our `_async` APIs with this macro, yes?

Reply:
> The cub/kernel call will immediately return and will not wait for the kernel to finish?

Yes

> Then we should probably wrap all kernel launches in our _async APIs with this macro, yes?

All cub and CUDA API invocations are guarded with `CUCO_CUDA_TRY`. If not, we need to add it. Kernels don't return CUDA error so no need to wrap them.

Reply:
Thanks for the clarification. Thread resolved :)